pkg/config/config.go

```diff
@@ -258,11 +302,24 @@ func ConfigureEcho() *echo.Echo {
 	e.Use(echo_middleware.RequestIDWithConfig(echo_middleware.RequestIDConfig{
 		TargetHeader: "x-rh-insights-request-id",
 	}))
-	e.Use(WrapMiddlewareWithSkipper(identity.EnforceIdentity, SkipLiveness))
 	e.Use(lecho.Middleware(lecho.Config{
 		Logger:       echoLogger,
 		RequestIDKey: "x-rh-insights-request-id",
 	}))
-	e.HTTPErrorHandler = CustomHTTPErrorHandler
+
+	return e
+}
+
+func ConfigureEchoMetrics() *echo.Echo {
+	e := echo.New()
+	configureEchoCommon(e)
+	return e
+}
+
+func ConfigureEchoService(metrics *instrumentation.Metrics) *echo.Echo {
+	e := echo.New()
+	e.Use(createMetricsMiddleware(metrics))
+	configureEchoCommon(e)
+	e.Use(WrapMiddlewareWithSkipper(identity.EnforceIdentity, SkipLiveness))
```

Comment:
Following up from my previous comment, I see you removed `metrics` from `configureEchoCommon`, which I think is a good simplification.

I still think you could make these functions even simpler, by replacing `configureEchoCommon`, `ConfigureEchoMetrics` and `ConfigureEchoService` with [these two functions](https://gist.github.com/rverdile/150de14374ee08d0d182daa16f8e5033).

If you don't like that, I think you should at least rename your current functions: `ConfigureEchoMetrics -> ConfigureEcho` and `ConfigureEchoService -> ConfigureEchoMetrics`. The current naming seems backwards.

Reply:
thanks! I like it! I will update with this! sorry if I missed it